webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/rest/internal/RestQuicksearch.scala

```diff
@@ -142,17 +142,19 @@ class RestQuicksearch(
    *   crunching the browser with thousands of answers
    */
 
-  private def prepare(results: Set[QuickSearchResult]): JValue = {
+  private def prepare(results: Set[QuickSearchResult], maxByKind: Int): JValue = {
     val filteredResult = filter(results)
     // group by kind, and build the summary for each
     val map            = filteredResult.groupBy(_.id.tpe).map {
       case (tpe, set) =>
-        // distinct by id:
-        val unique  = set.map(x => (x.id, x)).toMap.values.toSeq.sortBy(_.name)
-        // on take the nth first, sorted by name
-        val summary = ResultTypeSummary(tpe.name, unique.size, unique.size)
+        // distinct by id, sorted by name
+        val unique   = set.map(x => (x.id, x)).toMap.values.toSeq.sortBy(_.name)
+        // take the nth first
+        val returned = unique.take(maxByKind)
 
-        (tpe, (summary, unique))
+        val summary = ResultTypeSummary(tpe.name, unique.size, returned.size)
```

Comment:
still take the default behavior of MAX_RES_BY_KIND = 10 elements when no limit is specified, it is the supposed main usage of Quicksearch